debezium-testing/debezium-testing-testcontainers/src/main/java/io/debezium/testing/testcontainers/MongoDbReplicaSet.java

```diff
@@ -280,7 +280,7 @@ public void start() {
     public void stop() {
         LOGGER.info("[{}] Stopping...", name);
         MoreStartables.deepStopSync(members.stream());
-        network.close();
```

Comment:
I'd agree that the code was broken since it closed a potentially shared network when the database shut down. However the code is now broken in exactly the oposite way -- the code now doesn't close a network implicitely created by the builder, which is likely to be used when the user wants just the databases.

If we are fixing this it should be done properly. The network should be closed only if it was provided explicitely. In case the network is implicitly created by the builder (since you don't want to manually manage the only network in cases when database is all you need) then it should also be implicitly closed.

Reply:
I would agree that this can be causing issues, although I don't see that in the current test. Can you point me to an error message in the CI runs?
Personally I am still convinced, that if there's only one instance then the network would automatically been shutdown/closed at the end of the test / test suites run by testcontainers, no?
I like your idea but currently working tests show that it works that way and testcontainers might automatically remove the network at the end, doesn't it? And the added complexity might not be needed, is it  really required? I never saw such behavior anywhere else. Doesn't the GenericContainer create a network when none was configured? 
Furthermore, shouldn't another component exist that coordinates the whole test infrastructure/ test stack and for example connects the Mongo DB network to other components in the test stack? That components should manage this. I would personally remove the `Network.newNetwork()` code from the container code and let another component manage this.

Reply:
For all the tests but rest extension the entire "test stack" is just the databse.

Regardless if there the network is garbage collected -- if we are fixing something we should not break it in a different way. That's just good practice. I'm fine if you just open a Jira for this in order not to lose track and it can be fixed afterwards.

Reply:
That code line breaks tests that reuse the same MongoDB testcontainers like you would normally do when you are not coming from ancient old test stack of an aged project. So this line must go now and not later based on a Jira.

Reply:
Where did I state the oposite? I said that the fix is not complete and leaves a "TODO" if you will and as such should be captured in a followup jira. So please create one.

Reply:
Never mind. I misunderstood your previous comment then.

Reply:
I have logged https://issues.redhat.com/browse/DBZ-6948 to address this issue and also taking care of managing the `enableFakeDnsIfRequired()` related code. It should only be activated inside the container startup methods on a system environment that requires it instead of having this enabled somewhere outside. Alternatively this could be done with JUnit5 env-specific configuration on Mac.